### PR TITLE
Added `ParserAwareInterface` to expose `setParser()` method

### PR DESCRIPTION
**Pull request by MAXakaWIZARD**
Implements #50

### CODE REVIEW

**Comment by gonzofy** [comment]
@MAXakaWIZARD most if this is excellent and good to go. The one change is to switch from your tab changes back to spaces for whitespace to be consistent with the rest of the codebase.

**Comment by MAXakaWIZARD** [comment]
@gonzofy could you please point me to such example (tabs vs spaces)?

File: src/Listener/RegexListener.php [hunk 1]
```diff
@@ -1,269 +1,296 @@
...
      */
     const OBJECT = -1;
     /**
      * Constant to flag that it is an array level.
+     *
      * @var int
      */
     const ARRAY = 0;
-    
+
```
Comment (gonzofy) [inline src/Listener/RegexListener.php]: @MAXakaWIZARD example I believe.

Comment (MAXakaWIZARD) [reply src/Listener/RegexListener.php]: this one is removing extra whitespace at the line end

**Pull request merged**